### PR TITLE
Add Dynamic DeepHit (time-to-event)

### LINKED ISSUE #22: [Feat] Prediction models: Add Dynamic DeepHit
## Feature Description
Dynamic Deephit prediciton model.

Reference implementation in PyTorch: https://github.com/vanderschaarlab/synthcity/blob/main/src/synthcity/plugins/core/models/time_series_survival/ts_surv_dynamic_deephit.py

### LINKED ISSUE #34: Fix the problem of accessing static/class methods on plugins

### PR DESCRIPTION
**Pull request by DrShushen**
## Description
Closes #22

* Add `DynamicDeepHitTimeToEventAnalysis` and the infrastructure and models needed for that.
* Add time-to-event evaluation and benchmarking.

Separately:
* Make data loaders subclass `DataLoader` class.
* Rename some things to ensure consistent naming for the time-to-event concept.
* Remove some unused legacy code.
* Some additional data transformation helper.

## Affected Dependencies
`scikit-learn` added.

## How has this been tested?
- See tests, new additions all covered.

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/vanderschaarlab/.github/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/vanderschaarlab/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] I have commented my code following the [van der Schaar Lab Styleguide](https://github.com/vanderschaarlab/.github/blob/main/STYLEGUIDE.md)
- [X] I have labelled this PR with the relevant [Type labels](https://github.com/vanderschaarlab/.github/labels?q=Type%3A)
- [X] My changes are covered by tests

### CODE REVIEW

**Comment by DrShushen** [comment]
There are still some additional metrics that need to be added as per #9 - will add that in future PR

**Review by bcebere — APPROVED** [review]
Great work! LGTM!

There is a discussion about time-to-event naming:  There is dedicated literature for time-to-event regression, which approximates the date of the event. And is different from the current models you have.

Otherwise, survival models approximate the probability of risk(class 1) or survival(class 0) for multiple horizons. In AutoPrognosis I am approximating the probability of risk(same as here), so these models are named `risk_estimation` models.
